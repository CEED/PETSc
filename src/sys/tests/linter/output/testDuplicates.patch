@@ -9,5 +9,5 @@
   PetscAssertPointer(a, 1);
   /* remove */
-  PetscAssertPointer(a, 1);
+  
   /* no remove */
   PetscAssertPointer(b, 5);
@@ -11,5 +11,5 @@
   PetscAssertPointer(a, 1);
   /* no remove */
-  PetscAssertPointer(b, 5);
+  PetscAssertPointer(b, 2);
   /* ~should~ be removed but won't be */
   PetscAssertPointer(b, 7);
@@ -13,5 +13,5 @@
   PetscAssertPointer(b, 5);
   /* ~should~ be removed but won't be */
-  PetscAssertPointer(b, 7);
+  PetscAssertPointer(b, 2);
   PetscAssertPointer(b, 3);
   return 0;
@@ -14,5 +14,5 @@
   /* ~should~ be removed but won't be */
   PetscAssertPointer(b, 7);
-  PetscAssertPointer(b, 3);
+  PetscAssertPointer(b, 2);
   return 0;
 }
@@ -24,5 +24,5 @@
   PetscAssertPointer(b, 2);
   /* remove */
-  PetscAssertPointer(a, 1);
+  
   PetscAssertPointer(b, 2);
   {
@@ -25,5 +25,5 @@
   /* remove */
   PetscAssertPointer(a, 1);
-  PetscAssertPointer(b, 2);
+  
   {
     /* remove */
@@ -28,5 +28,5 @@
   {
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
   }
@@ -29,5 +29,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
   }
   return 0;
@@ -40,5 +40,5 @@
   PetscAssertPointer(b, 2);
   /* remove */
-  PetscAssertPointer(a, 1);
+  
   PetscAssertPointer(b, 2);
   {
@@ -41,5 +41,5 @@
   /* remove */
   PetscAssertPointer(a, 1);
-  PetscAssertPointer(b, 2);
+  
   {
     /* remove */
@@ -44,5 +44,5 @@
   {
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
   }
@@ -45,5 +45,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
   }
   {
@@ -49,5 +49,5 @@
   {
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
   }
@@ -50,5 +50,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
   }
   return 0;
@@ -80,5 +80,5 @@
   case PETSC_TRUE:
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
     break;
@@ -81,5 +81,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
     break;
   case PETSC_FALSE:
@@ -85,5 +85,5 @@
   case PETSC_FALSE:
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
     break;
@@ -86,5 +86,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
     break;
   }
@@ -113,5 +113,5 @@
   if (cond) {
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
   } else {
@@ -114,5 +114,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
   } else {
     /* remove */
@@ -117,5 +117,5 @@
   } else {
     /* remove */
-    PetscAssertPointer(a, 1);
+    
     PetscAssertPointer(b, 2);
   }
@@ -118,5 +118,5 @@
     /* remove */
     PetscAssertPointer(a, 1);
-    PetscAssertPointer(b, 2);
+    
   }
   return 0;
