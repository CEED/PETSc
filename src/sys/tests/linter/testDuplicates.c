#include <petsc/private/petscimpl.h>

PetscErrorCode testDuplicatesWithChanges(PetscInt *a, PetscScalar *b)
{
  /* no remove */
  PetscValidPointer(a, 1);
  /* remove */
  PetscValidPointer(a, 1);
  /* no remove */
  PetscValidPointer(b, 5);
  /* ~should~ be removed but won't be */
  PetscValidPointer(b, 7);
  PetscValidPointer(b, 3);
  return 0;
}

PetscErrorCode testDuplicatesScoped(PetscInt *a, PetscScalar *b)
{
  /* no remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  /* remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  {
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  return 0;
}

PetscErrorCode testDuplicatesDoubleScoped(PetscInt *a, PetscScalar *b)
{
  /* no remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  /* remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  {
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  {
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  return 0;
}

PetscErrorCode testNoDuplicatesSwitch(PetscInt *a, PetscScalar *b, PetscBool cond)
{
  switch (cond) {
  case PETSC_TRUE:
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
    break;
  case PETSC_FALSE:
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
    break;
  }
  return 0;
}

PetscErrorCode testDuplicatesNoChangesSwitch(PetscInt *a, PetscScalar *b, PetscBool cond)
{
  /* no remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  switch (cond) {
  case PETSC_TRUE:
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
    break;
  case PETSC_FALSE:
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
    break;
  }
  return 0;
}

PetscErrorCode testNoDuplicatesIfElse(PetscInt *a, PetscScalar *b, PetscBool cond)
{
  if (cond) {
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  } else {
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  return 0;
}

PetscErrorCode testDuplicatesIfElse(PetscInt *a, PetscScalar *b, PetscBool cond)
{
  /* no remove */
  PetscValidPointer(a, 1);
  PetscValidPointer(b, 2);
  if (cond) {
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  } else {
    /* remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  return 0;
}

PetscErrorCode testNoDuplicatesIfElseIfElse(PetscInt *a, PetscScalar *b, PetscBool cond)
{
  if (cond) {
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  } else if (!cond) {
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  } else {
    /* no remove */
    PetscValidPointer(a, 1);
    PetscValidPointer(b, 2);
  }
  return 0;
}
