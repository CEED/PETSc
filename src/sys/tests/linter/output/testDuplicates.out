<--- FIXED --->
[31m[1m./src/sys/tests/linter/testDuplicates.c:10:25: error:[0m Argument number doesn't match for 'b'. Expected '2', found '5' [-fmatching-arg-num]:
  8 : PetscAssertPointer(a, 1);
  9 : /* no remove */
> 10: PetscAssertPointer(b, [33m[1m5[0m);
                            [33m[1m^[0m
  11: /* ~should~ be removed but won't be */
  12: PetscAssertPointer(b, 7);

./src/sys/tests/linter/testDuplicates.c:3:55 Note: 'b' is traceable to argument #2 'b' in enclosing function here:
  1: #include <petsc/private/petscimpl.h>
  2: 
> 3: PetscErrorCode testDuplicatesWithChanges(PetscInt *a, [33m[1mPetscScalar *b[0m)
                                                           [33m[1m^^^^^^^^^^^^^^[0m
  4: {
  5:   /* no remove */
[31m[1m./src/sys/tests/linter/testDuplicates.c:12:25: error:[0m Argument number doesn't match for 'b'. Expected '2', found '7' [-fmatching-arg-num]:
  10: PetscAssertPointer(b, 5);
  11: /* ~should~ be removed but won't be */
> 12: PetscAssertPointer(b, [33m[1m7[0m);
                            [33m[1m^[0m
  13: PetscAssertPointer(b, 3);
  14: return 0;

./src/sys/tests/linter/testDuplicates.c:3:55 Note: 'b' is traceable to argument #2 'b' in enclosing function here:
  1: #include <petsc/private/petscimpl.h>
  2: 
> 3: PetscErrorCode testDuplicatesWithChanges(PetscInt *a, [33m[1mPetscScalar *b[0m)
                                                           [33m[1m^^^^^^^^^^^^^^[0m
  4: {
  5:   /* no remove */
[31m[1m./src/sys/tests/linter/testDuplicates.c:13:25: error:[0m Argument number doesn't match for 'b'. Expected '2', found '3' [-fmatching-arg-num]:
  11:   /* ~should~ be removed but won't be */
  12:   PetscAssertPointer(b, 7);
> 13:   PetscAssertPointer(b, [33m[1m3[0m);
                              [33m[1m^[0m
  14:   return 0;
  15: }

./src/sys/tests/linter/testDuplicates.c:3:55 Note: 'b' is traceable to argument #2 'b' in enclosing function here:
  1: #include <petsc/private/petscimpl.h>
  2: 
> 3: PetscErrorCode testDuplicatesWithChanges(PetscInt *a, [33m[1mPetscScalar *b[0m)
                                                           [33m[1m^^^^^^^^^^^^^^[0m
  4: {
  5:   /* no remove */
[31m[1m./src/sys/tests/linter/testDuplicates.c:8:3: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  4: {
  5:   /* no remove */
> 6:   [33m[1mPetscAssertPointer(a, 1)[0m;
       [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  7:   /* remove */
  8:   PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:23:3: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  18: {
  19:   /* no remove */
> 20:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  21:   PetscAssertPointer(b, 2);
  22:   /* remove */
  23:   PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:24:3: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  19: /* no remove */
  20: PetscAssertPointer(a, 1);
> 21: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  22: /* remove */
  23: PetscAssertPointer(a, 1);
  24: PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:27:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  18: {
  19:   /* no remove */
> 20:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  21:   PetscAssertPointer(b, 2);
  22:   /* remove */
  23:   PetscAssertPointer(a, 1);
  24:   PetscAssertPointer(b, 2);
  25:   {
  26:     /* remove */
  27:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:28:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  19: /* no remove */
  20: PetscAssertPointer(a, 1);
> 21: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  22: /* remove */
  23: PetscAssertPointer(a, 1);
  24: PetscAssertPointer(b, 2);
  25: {
  26:   /* remove */
  27:   PetscAssertPointer(a, 1);
  28:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:39:3: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  34: {
  35:   /* no remove */
> 36:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  37:   PetscAssertPointer(b, 2);
  38:   /* remove */
  39:   PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:40:3: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  35: /* no remove */
  36: PetscAssertPointer(a, 1);
> 37: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  38: /* remove */
  39: PetscAssertPointer(a, 1);
  40: PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:43:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  34: {
  35:   /* no remove */
> 36:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  37:   PetscAssertPointer(b, 2);
  38:   /* remove */
  39:   PetscAssertPointer(a, 1);
  40:   PetscAssertPointer(b, 2);
  41:   {
  42:     /* remove */
  43:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:44:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  35: /* no remove */
  36: PetscAssertPointer(a, 1);
> 37: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  38: /* remove */
  39: PetscAssertPointer(a, 1);
  40: PetscAssertPointer(b, 2);
  41: {
  42:   /* remove */
  43:   PetscAssertPointer(a, 1);
  44:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:48:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  34: {
  35:   /* no remove */
> 36:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  37:   PetscAssertPointer(b, 2);
  38:   /* remove */
  39:   PetscAssertPointer(a, 1);
  40:   PetscAssertPointer(b, 2);
  41:   {
  42:     /* remove */
  43:     PetscAssertPointer(a, 1);
  44:     PetscAssertPointer(b, 2);
  45:   }
  46:   {
  47:     /* remove */
  48:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:49:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  35: /* no remove */
  36: PetscAssertPointer(a, 1);
> 37: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  38: /* remove */
  39: PetscAssertPointer(a, 1);
  40: PetscAssertPointer(b, 2);
  41: {
  42:   /* remove */
  43:   PetscAssertPointer(a, 1);
  44:   PetscAssertPointer(b, 2);
  45: }
  46: {
  47:   /* remove */
  48:   PetscAssertPointer(a, 1);
  49:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:79:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  72: {
  73:   /* no remove */
> 74:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  75:   PetscAssertPointer(b, 2);
  76:   switch (cond) {
  77:   case PETSC_TRUE:
  78:     /* remove */
  79:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:80:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  73: /* no remove */
  74: PetscAssertPointer(a, 1);
> 75: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  76: switch (cond) {
  77: case PETSC_TRUE:
  78:   /* remove */
  79:   PetscAssertPointer(a, 1);
  80:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:84:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  72: {
  73:   /* no remove */
> 74:   [33m[1mPetscAssertPointer(a, 1)[0m;
        [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  75:   PetscAssertPointer(b, 2);
  76:   switch (cond) {
  77:   case PETSC_TRUE:
  78:     /* remove */
  79:     PetscAssertPointer(a, 1);
  80:     PetscAssertPointer(b, 2);
  81:     break;
  82:   case PETSC_FALSE:
  83:     /* remove */
  84:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:85:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  73: /* no remove */
  74: PetscAssertPointer(a, 1);
> 75: [33m[1mPetscAssertPointer(b, 2)[0m;
      [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  76: switch (cond) {
  77: case PETSC_TRUE:
  78:   /* remove */
  79:   PetscAssertPointer(a, 1);
  80:   PetscAssertPointer(b, 2);
  81:   break;
  82: case PETSC_FALSE:
  83:   /* remove */
  84:   PetscAssertPointer(a, 1);
  85:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:112:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  106: {
  107:   /* no remove */
> 108:   [33m[1mPetscAssertPointer(a, 1)[0m;
         [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  109:   PetscAssertPointer(b, 2);
  110:   if (cond) {
  111:     /* remove */
  112:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:113:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  107: /* no remove */
  108: PetscAssertPointer(a, 1);
> 109: [33m[1mPetscAssertPointer(b, 2)[0m;
       [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  110: if (cond) {
  111:   /* remove */
  112:   PetscAssertPointer(a, 1);
  113:   PetscAssertPointer(b, 2);
[31m[1m./src/sys/tests/linter/testDuplicates.c:116:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  106: {
  107:   /* no remove */
> 108:   [33m[1mPetscAssertPointer(a, 1)[0m;
         [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  109:   PetscAssertPointer(b, 2);
  110:   if (cond) {
  111:     /* remove */
  112:     PetscAssertPointer(a, 1);
  113:     PetscAssertPointer(b, 2);
  114:   } else {
  115:     /* remove */
  116:     PetscAssertPointer(a, 1);
[31m[1m./src/sys/tests/linter/testDuplicates.c:117:5: error:[0m Duplicate function found previous identical usage [-fduplicate-function]:
  107: /* no remove */
  108: PetscAssertPointer(a, 1);
> 109: [33m[1mPetscAssertPointer(b, 2)[0m;
       [33m[1m^^^^^^^^^^^^^^^^^^^^^^^^[0m
  110: if (cond) {
  111:   /* remove */
  112:   PetscAssertPointer(a, 1);
  113:   PetscAssertPointer(b, 2);
  114: } else {
  115:   /* remove */
  116:   PetscAssertPointer(a, 1);
  117:   PetscAssertPointer(b, 2);
<--- LEFT --->
